crates/rpc/rpc-builder/tests/it/http.rs

```diff
@@ -99,15 +99,23 @@ where
     EthApiClient::transaction_by_hash(client, tx_hash).await.unwrap();
     EthApiClient::transaction_by_block_hash_and_index(client, hash, index).await.unwrap();
     EthApiClient::transaction_by_block_number_and_index(client, block_number, index).await.unwrap();
-    EthApiClient::create_access_list(client, call_request.clone(), Some(block_number.into()))
-        .await
-        .unwrap();
-    EthApiClient::estimate_gas(client, call_request.clone(), Some(block_number.into()))
-        .await
-        .unwrap();
-    EthApiClient::call(client, call_request.clone(), Some(block_number.into()), None, None)
-        .await
-        .unwrap();
+    let access_list_result =
+        EthApiClient::create_access_list(client, call_request.clone(), Some(block_number.into()))
+            .await;
+    if let Err(err) = access_list_result {
+        assert!(err.to_string().contains("blob transaction missing blob hashes"));
+    }
+    let estimate_gas_result =
+        EthApiClient::estimate_gas(client, call_request.clone(), Some(block_number.into())).await;
+    if let Err(err) = estimate_gas_result {
+        assert!(err.to_string().contains("blob transaction missing blob hashes"));
+    }
+    let call_result =
+        EthApiClient::call(client, call_request.clone(), Some(block_number.into()), None, None)
+            .await;
+    if let Err(err) = call_result {
+        assert!(err.to_string().contains("blob transaction missing blob hashes"));
+    }
```

Comment:
why are these failing?
they should not fail, there are no blobs

Reply:
If u see in the ensure_fees function https://github.com/supernovahs/reth/blob/4586be428cc453ebb924b49c175f96af4da35017/crates/rpc/rpc/src/eth/revm_utils.rs#L377

we are giving value of `max_fee_per_blob_gas` as a Some value by default. (In tests, I checked using print)
So the third arm i.e 4844 , matches. 
And if u  print `blob_versioned_hashes`, it is None.
So as per out logic , we should revert right?

Inserted this line at the start of ensure_fees function
`        println!("ensure fee params,{:?},{:?}",max_fee_per_blob_gas.is_none(),blob_versioned_hashes);
`

Reply:
> we are giving value of max_fee_per_blob_gas as a Some value by default

ah sigh, this breaks our logic and would leads to these sideffects that renders the call invalid...

need to think about this a bit, otherwise this pr is complete

Reply:
ok